### ISSUE TITLE
Service Discovery WATO Exception

### ISSUE THREAD

**Issue by ivmiddj** [comnetgmbh/check_elasticsearch#2]
Hallo,

beim Service Discovery im Wato erhalte ich folgende Meldung:

Service discovery failed for this host: Got invalid data:

Exception in check plugin 'elasticsearch_health': need more than 1 value to unpack

![bildschirmfoto 2018-05-07 um 11 48 14](https://user-images.githubusercontent.com/39051015/39696372-da4d2c14-51ed-11e8-8bab-f49b8fa14f96.png)

Viele Grüsse,

Jörg

**Comment by admlko** [comment]
I can confirm this.

There seems to be an issue with inventory function, for example inventory_elasticsearch_health:
```
def inventory_elasticsearch_health(info):
    for title, value in info:
```

If I print the content of info:
`[[u'status;green'], [u'number_of_nodes;3'], [u'unassigned_shards;0'], [u'number_of_pending_tasks;0'], [u'number_of_in_flight_fetch;0'], [u'timed_out;False'], [u'active_primary_shards;1000'], [u'task_max_waiting_in_queue_millis;0'], [u'cluster_name;prod'], [u'relocating_shards;0'], [u'active_shards_percent_as_number;100.0'], [u'active_shards;1000'], [u'initializing_shards;0'], [u'number_of_data_nodes;3'], [u'delayed_unassigned_shards;0']]
`

And if I try to loop over that with `for title, value in info:` I get exactly that error: 
> ValueError: need more than 1 value to unpack

I am not sure where to start fixing the issue, so let's hear from the developer first :)

**Comment by veglavas** [comment]
From what I've found, the problem is with the agent plugin. It outputs the values separated by semicolons. If you change every semicolon in section printing to a space, Check_MK properly parses the output. For instance:
`[u'status;green']`
becomes
`[u'status', u'green']`

Cheers,
Vedran

**Comment by mludeke** [comment]
You have to specify the separator ("`:sep(59)`" for ascii char semicolon) in the output of the agent before "`>>>`"

Unfortunately, this option is not (yet) documented.

You have to change the following in `/usr/lib/check_mk_agent/plugins/elasticsearch`

--- a/elasticsearch       2018-06-19 17:40:17.509432166 +0200
+++ b/elasticsearch       2018-06-21 23:45:42.764140078 +0200
@@ -73,7 +73,7 @@
     response = requests.get(health_url)
     elasticsearch_health_response = response.json()
 
-    print "<<<elasticsearch_health>>>"
+    print "<<<elasticsearch_health:sep(59)>>>"
     for item in elasticsearch_health_response:
         print "%s;%s" % (item, elasticsearch_health_response[item])
 
@@ -83,7 +83,7 @@
     response = requests.get(nodestats_url)
     elasticsearch_nodestats_response = response.json()
 
-    print "<<<elasticsearch_nodestats>>>"
+    print "<<<elasticsearch_nodestats:sep(59)>>>"
     for node in elasticsearch_nodestats_response["nodes"]:
         node = elasticsearch_nodestats_response["nodes"][node]
         node_name = node["name"]
@@ -100,19 +100,19 @@
     response = requests.get(stats_url)
     elasticsearch_stats_response = response.json()
 
-    print "<<<elasticsearch_shards>>>"
+    print "<<<elasticsearch_shards:sep(59)>>>"
     shards_total = elasticsearch_stats_response["_shards"]["total"]
     shards_success = elasticsearch_stats_response["_shards"]["successful"]
     shards_failed = elasticsearch_stats_response["_shards"]["failed"]
     print "%s;%s;%s" % (shards_total, shards_success, shards_failed)
 
-    print "<<<elasticsearch_cluster>>>"
+    print "<<<elasticsearch_cluster:sep(59)>>>"
     if "docs" in elasticsearch_stats_response["_all"]["total"]:
         count = elasticsearch_stats_response["_all"]["total"]["docs"]["count"]
         size = elasticsearch_stats_response["_all"]["total"]["store"]["size_in_bytes"]
         print "%s;%s" % (count, size)
 
-    print "<<<elasticsearch_indices>>>"
+    print "<<<elasticsearch_indices:sep(59)>>>"
     indices = set()
     for i in elasticsearch_stats_response["indices"]:
         name = i[:-11]

Regards
Mirko

**Comment by RikaDenia** [comment]
Fixed by #4
